Add service worker logging to ease debugging of missing service workers/missing event processing

youennf opened the pull request:
#### b39f7b61225c26bb90158a196fc0e1e1b54b0880
<pre>
Add service worker logging to ease debugging of missing service workers/missing event processing
<a href="https://bugs.webkit.org/show_bug.cgi?id=256890">https://bugs.webkit.org/show_bug.cgi?id=256890</a>
rdar://109455831

Reviewed by Ben Nham.

No functional change except additional and/or more precise service worker related logging.

* Source/WebCore/workers/service/context/SWContextManager.cpp:
(WebCore::SWContextManager::fireInstallEvent):
(WebCore::SWContextManager::fireActivateEvent):
(WebCore::SWContextManager::firePushSubscriptionChangeEvent):
(WebCore::SWContextManager::terminateWorker):
* Source/WebCore/workers/service/server/SWServer.cpp:
(WebCore::SWServer::addContextConnection):
(WebCore::SWServer::removeContextConnection):
(WebCore::SWServer::fireFunctionalEvent):

Canonical link: <a href="https://commits.webkit.org/264411@main">https://commits.webkit.org/264411@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/def54ee76c22f2d9f7b5b5a450f14a2cfcf19b73

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/7388 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/7646 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/7821 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/9017 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/7593 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/7398 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/9493 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/7571 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/10478 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/7517 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/8573 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/6836 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/9126 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/5885 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/6720 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/14446 "1 flakes 133 failures") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/7182 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/6853 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/10081 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/7322 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/5986 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/6674 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1793 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/10880 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/7058 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

bnham reviewed (approved):
Seems fine to me, but should we also add this logging to the other events in the file? For instance, `firePushEvent`, `fireNotificationEvent`, ...

youennf committed:
Add service worker logging to ease debugging of missing service workers/missing event processing
https://bugs.webkit.org/show_bug.cgi?id=256890
rdar://109455831

Reviewed by Ben Nham.

No functional change except additional and/or more precise service worker related logging.

* Source/WebCore/workers/service/context/SWContextManager.cpp:
(WebCore::SWContextManager::fireInstallEvent):
(WebCore::SWContextManager::fireActivateEvent):
(WebCore::SWContextManager::firePushSubscriptionChangeEvent):
(WebCore::SWContextManager::terminateWorker):
* Source/WebCore/workers/service/server/SWServer.cpp:
(WebCore::SWServer::addContextConnection):
(WebCore::SWServer::removeContextConnection):
(WebCore::SWServer::fireFunctionalEvent):

Canonical link: https://commits.webkit.org/264411@main
